### PR TITLE
Add new Xtext editor quickfixes on *.xtext files

### PR DESCRIPTION
**Pull request by nbhusare**
Signed-off-by: nbhusare <neerajbhusare@gmail.com>

### CODE REVIEW

File: org.eclipse.xtext/src/org/eclipse/xtext/xtext/XtextValidator.java [hunk 1]
```diff
@@ -810,7 +810,7 @@ public void checkTerminalFragmentCalledFromTerminalRule(final RuleCall call) {
 							call, 
 							XtextPackage.Literals.RULE_CALL__RULE,
 							ValidationMessageAcceptor.INSIGNIFICANT_INDEX,
-							null);
+							ILLEGAL_RULE_REFERENCE);
```
Comment (szarnekow) [inline org.eclipse.xtext/src/org/eclipse/xtext/xtext/XtextValidator.java:813]: There are other validations, that appear to do similar validation, e.g. in line 796. Please use a more specific field name or provide an issue code for the other cases, too.

File: org.eclipse.xtext/src/org/eclipse/xtext/xtext/XtextConfigurableIssueCodes.java [hunk 2]
```diff
@@ -44,6 +44,7 @@
 	public static final String BIDIRECTIONAL_REFERENCE = ISSUE_CODE_PREFIX + "BidirectionalReference";
 	public static final String CROSS_REFERENCE_IN_ALTERNATIVES = ISSUE_CODE_PREFIX + "CrossReferenceInAlternatives";
 	public static final String INVALID_FRAGMENT_AS_FIRST_RULE = ISSUE_CODE_PREFIX + "InvalidFragmentFirstRule";
+	public static final String ILLEGAL_RULE_REFERENCE = ISSUE_CODE_PREFIX + "InvalidTerminalRuleCall";
```
Comment (szarnekow) [inline org.eclipse.xtext/src/org/eclipse/xtext/xtext/XtextConfigurableIssueCodes.java:47]: Maybe ILLEGAL_RULE_REFERENCE_IN_TERMINAL?

**Comment by cdietrich** [comment]
@szarnekow @nbhusare what is the current state of this PR?

**Comment by nbhusare** [comment]
@cdietrich I'll be pushing the updated code today.

**Comment by nbhusare** [comment]
Raised another PR as I was facing some issues with this branch - https://github.com/eclipse/xtext-core/pull/1540

**Pull request closed without merging**